### PR TITLE
Cc 54 dynamic back navigation

### LINKED ISSUE #54: Dynamic Back Navigation

### PR DESCRIPTION
**Pull request by czcarrizales**
## Changes
_List Changes Introduced by this PR_
1. Created navigation service, to allow users to go back to previous page
2. Added back buttons to game and online-match components

Closes #54

### CODE REVIEW

**Comment by czcarrizales** [comment]
So, I made a pull request, only to see what you all think about making a navigation service. Also, I will make sure to clean up the look of the buttons, I just wanted you all to test how it work, and see if that aligns with the project.

File: src/client/src/app/services/navigation.service.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { Location } from "@angular/common";
+import { Router, NavigationEnd } from "@angular/router";
+
+@Injectable({providedIn: 'root'})
+export class NavigationService{
+    private history: string[] = []
+
+    constructor(private router: Router, private location: Location) {
+        this.router.events.subscribe((event) => {
+            if (event instanceof NavigationEnd) {
```
Comment (ChelleAmon) [inline src/client/src/app/services/navigation.service.ts:11]: I tried to console log the history to see how the router events work I guess but I am still kind of lost on how this works lol. Just want to put a comment here so you will see and maybe explain to me what this does for my learning?

**Review by ChelleAmon — APPROVED** [review]
The Back navigation functionality works as intended. I like that you add a navigation service to handle any navigation logic. I left a comment about the NavigationEnd event because I am still confused about it (Documentation about it does not make sense to me right now lol) and maybe you can explain it to me just for my personal learning. Still going to approve these changes and you can merge after resolving the conversation

File: src/client/src/app/services/navigation.service.ts (hunk 1 above)
Comment (ChelleAmon) [reply src/client/src/app/services/navigation.service.ts:11]: Going to resolve this. Anyways, this is only a question that can be researched later on.

**Pull request merged**